motionplan/armplanning/node.go

```diff
@@ -128,7 +128,7 @@ type solutionSolvingState struct {
 }
 
 func newSolutionSolvingState(ctx context.Context, psc *planSegmentContext) (*solutionSolvingState, error) {
-	_, span := trace.StartSpan(ctx, "newSolutionSolvingState")
+	ctx, span := trace.StartSpan(ctx, "newSolutionSolvingState")
```

Comment:
This is super subtle. By changing `newSolutionSolvingState` to actually use its context beyond tracing, forgetting to re-assign ctx here creates a bug.

I figured this was Go itself refusing to compile because of "unused variable". But now I see Go itself is smart enough to not complain about this and it's our linter instead:
```
motionplan/armplanning/node.go:131:2: ineffectual assignment to ctx (ineffassign)
	ctx, span := trace.StartSpan(ctx, "newSolutionSolvingState")
```